### PR TITLE
Simplify rank badges and implement profile tags

### PR DESCRIPTION
**Pull request by sudokoko**
This PR simplifies the website rank badges to use existing classes, and also implements a small feature that allows server administrators to grant vanity profile tags to users, i.e. for winning an event.

Screenshot 2023-08-26 at 9 35 14 PM

### CODE REVIEW

**Review by Zaprit — APPROVED** [review]
LGTM

File: ProjectLighthouse.Servers.Website/Pages/UserSettingsPage.cshtml.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public UserSettingsPage(DatabaseContext database) : base(database)
     {}
 
     [SuppressMessage("ReSharper", "SpecifyStringComparison")]
-    public async Task<IActionResult> OnPost([FromRoute] int userId, [FromForm] string? avatar, [FromForm] string? username, [FromForm] string? email, [FromForm] string? biography, [FromForm] string? timeZone, [FromForm] string? language)
+    public async Task<IActionResult> OnPost([FromRoute] int userId, [FromForm] string? avatar, [FromForm] string? username, [FromForm] string? email, [FromForm] string profileTag, [FromForm] string? biography, [FromForm] string? timeZone, [FromForm] string? language)
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/UserSettingsPage.cshtml.cs:23]: These parameters should all probably be on separate lines. They were pretty long before but with another field it should probably be wrapped.

File: ProjectLighthouse.Servers.Website/Pages/UserSettingsPage.cshtml.cs [hunk 2]
```diff
@@ -33,6 +33,8 @@ public async Task<IActionResult> OnPost([FromRoute] int userId, [FromForm] strin
 
         if (avatarHash != null) this.ProfileUser.IconHash = avatarHash;
 
+        if (this.User.IsAdmin) this.ProfileUser.ProfileTag = profileTag;
```
Comment (Slendy) [inline ProjectLighthouse.Servers.Website/Pages/UserSettingsPage.cshtml.cs:36]: Does this also need to check if profile tag isn't null?

**Review by Slendy — CHANGES_REQUESTED** [review]
Mostly LGTM just some small concerns

File: ProjectLighthouse.Servers.Website/Pages/UserSettingsPage.cshtml.cs (hunk 2 above)
Comment (sudokoko) [reply ProjectLighthouse.Servers.Website/Pages/UserSettingsPage.cshtml.cs:36]: No - if it's null or whitespace it just isn't rendered

**Review by Slendy — APPROVED** [review]

**Pull request merged**